py/desispec/workflow/procfuncs.py

```diff
@@ -803,21 +802,53 @@ def checkfor_and_submit_joint_job(ptable, arcs, flats, sciences, arcjob, flatjob
         internal_id, int, if no job is submitted, this is the same as the input, otherwise it is incremented upward from
                           from the input such that it represents the smallest unused ID.
     """
-    if lasttype == 'science' and last_not_dither:
+    if lasttype == 'science':
+        log = get_logger()
+        skysubonly = np.array([sci['LASTSTEP'] == 'skysub' for sci in sciences])
+        if np.all(skysubonly):
+            log.info("Identified all exposures in joint fitting request as skysub-only. Not submitting")
+            sciences = []
+            return ptable, arcjob, flatjob, sciences, internal_id
+
+        if np.any(skysubonly):
+            log.info("Identified skysub-only exposures in joint fitting request")
+            log.info("Expid's: {}".format([row['EXPID'] for row in sciences]))
+            log.info("LASTSTEP's: {}".format([row['LASTSTEP'] for row in sciences]))
+            sciences = (np.array(sciences,dtype=object)[~skysubonly]).tolist()
+            log.info("Removed skysub only exposures in joint fitting:")
+            log.info("Expid's: {}".format([row['EXPID'] for row in sciences]))
+            log.info("LASTSTEP's: {}".format([row['LASTSTEP'] for row in sciences]))
+
+        from collections import Counter
+        tiles = np.array([sci['TILEID'] for sci in sciences])
+        counts = Counter(tiles)
+        if len(counts.most_common()) > 1:
+            log.info("Identified more than one tile in a joint fitting request")
+            log.info("Expid's: {}".format([row['EXPID'] for row in sciences]))
+            log.info("Tileid's: {}".format(tiles))
+            most_common, nmost_common = counts.most_common()[0]
+            if most_common == -99:
+                most_common, nmost_common = counts.most_common()[1]
+            log.warning(f"Given multiple tiles to jointly fit: {counts}. Only processing the most common non-default" +
+                        f" tile: {most_common} with {nmost_common} exposures")
+            sciences = (np.array(sciences,dtype=object)[tiles == most_common]).tolist()
+            log.info("Tiles and exposure id's being submitted for joint fitting:")
+            log.info("Expid's: {}".format([row['EXPID'] for row in sciences]))
+            log.info("Tileid's: {}".format([row['TILEID'] for row in sciences]))
```

Comment:
Having some but not all LASTSTEP=skysub and having more than one tile in the group are both error conditions that shouldn't occur if everything else is working correctly, right?  Here you make it robust and proceed, but if this case really isn't supposed to occur, perhaps we should print a message and exit so that we can fix the underlying problem that led to it?

Reply:
It's a great point and I thought about this when I was developing it. If you feel strongly I can remove it, but below is my thought process:

It's true that if we're sub-selecting from the list then something in the workflow logic went wrong. The goal of this robustness is that if it fails in a similar mechanistic way to the failure that spurred this PR -- but obviously through a different chain of logic issues -- it would still do the right thing in that circumstance. It's effectively an independent solution to the initial bug that is more general and could theoretically apply to a different unknown problem. 

This protects against bookkeeping issues where e.g. an exposure was added to the list accidentally or the joint fitting wasn't triggered and so two tiles worth of exposures were left in the list until the next triggering event. In either of those cases, this will perform a correct action and process at least one of the tiles. It may not process both if two tiles were lumped together, but it goes part-way and at least does one of them. It also throws up a log WARNING that is grep-able and provides a lot of output to work backward through its decision-making.

There could certainly be an unknown-unknown situation in which this generic catch-all would plod forward with running a subselection and that would be the incorrect thing to do. In that case, we'd have to delete those files and clean up by hand. The only difference between that situation and the situation created by not running at all is in the processing time used on the useless outputs and the effort of removing the bad output files. Recall that these post-extraction steps are quite quick from a processing perspective. 

Given that I can't imagine one of the unknown-unknown scenarios in which that would occur due to a breakdown in logic, whereas I can imagine failures that would be caught, I think the net advantage is to plod on.